libs/math/tests/unit/tensor/constructor.cpp

```diff
@@ -37,6 +37,144 @@ TYPED_TEST(TensorConstructorTest, string_construction)
   ASSERT_EQ(tensor.At(0, 0), TypeParam(1));
   ASSERT_EQ(tensor.At(0, 1), TypeParam(3));
   ASSERT_EQ(tensor.At(0, 2), TypeParam(4));
+
+  auto tensor1 = fetch::math::Tensor<TypeParam>::FromString(R"(
+  	1 3 4 4.1
+  	5 6 7 7.999
+  	8 9 10 11.11111
+  )");
+  ASSERT_EQ(tensor1.shape().at(0), 3);
+  ASSERT_EQ(tensor1.shape().at(1), 4);
+  ASSERT_EQ(tensor1.At(0, 0), math::Type<TypeParam>("1"));
```

Comment:
(optional) ASSERT_EQ will immediately kill the test if failed, but each comparison here and in next line is actually not so critical - if it fails, we still could check the rest of elements; so I would suggest EXPECT_EQ for element comparisons.

Reply:
replaced with EXPECT